addons/mrp/models/mrp_workcenter.py

```diff
@@ -75,13 +79,69 @@ class MrpWorkcenter(models.Model):
     tag_ids = fields.Many2many('mrp.workcenter.tag')
     capacity_ids = fields.One2many('mrp.workcenter.capacity', 'workcenter_id', string='Product Capacities',
         help="Specific number of pieces that can be produced in parallel per product.", copy=True)
+    kanban_dashboard_graph = fields.Text(compute='_compute_kanban_dashboard_graph')
 
     @api.constrains('alternative_workcenter_ids')
     def _check_alternative_workcenter(self):
         for workcenter in self:
             if workcenter in workcenter.alternative_workcenter_ids:
                 raise ValidationError(_("Workcenter %s cannot be an alternative of itself.", workcenter.name))
 
+    def _compute_kanban_dashboard_graph(self):
+
+        def _get_date_array():
+            date_array = {}
+            today = datetime.today()
+            locale = get_lang(self.env).code
+            for delta in range(-1, 4):
+                delta_date = today + relativedelta.relativedelta(weeks=delta)
+                week_start = start_of(delta_date, 'week')
+                week_end = end_of(delta_date, 'week')
+                short_name = (format_date(week_start, 'd - ', locale=locale)
+                              + format_date(week_end, 'd MMM', locale=locale))
+                if not delta:
+                    short_name = 'This Week'
+                date_array[(week_start.strftime('%Y-%m-%d'), week_end.strftime('%Y-%m-%d'))] = short_name
+            return date_array
+
+        week_range = _get_date_array()
+        load_data = self._get_workcenter_load_per_week(week_range)
+        load_graph_data = self._prepare_graph_data(load_data, week_range)
+        for wc in self:
+            wc.kanban_dashboard_graph = json.dumps(load_graph_data[wc.id])
+
+    def _get_workcenter_load_per_week(self, week_range):
+        load_data = {rec: {} for rec in self}
+        if not self.order_ids:
+            # demo data
+            for wc in self:
+                load_limit = sum(wc.resource_calendar_id.attendance_ids.mapped('duration_hours'))
```

Comment:
A bit overkill for sample records no? I would just use a randint with a "valid" range

Reply:
fixed, it's now 40 since it's the default max load in hours for a new workcenter on a new DB. After that, there should be MOs so it won't matter